nodejs/eks/nodegroup.ts

```diff
@@ -829,6 +829,59 @@ function createNodeGroupInternal(
             }
         });
 
+    const amiInfo = pulumi.output(amiId).apply((id) =>
+        aws.ec2.getAmi(
+            {
+                owners: ["self", "amazon"],
+                filters: [
+                    {
+                        name: "image-id",
+                        values: [id],
+                    },
+                ],
+            },
+            { parent },
+        ),
+    );
+
+    const { rootBlockDevice, ebsBlockDevices } = pulumi
+        .all([os, amiInfo])
+        .apply(([os, amiInfo]) => {
+            if (os !== OperatingSystem.Bottlerocket) {
+                return {
+                    rootBlockDevice: {
+                        encrypted: args.nodeRootVolumeEncrypted ?? false,
+                        volumeSize: args.nodeRootVolumeSize ?? 20, // GiB
+                        volumeType: args.nodeRootVolumeType ?? "gp2",
+                        iops: args.nodeRootVolumeIops,
+                        throughput: args.nodeRootVolumeThroughput,
+                        deleteOnTermination: args.nodeRootVolumeDeleteOnTermination ?? true,
+                    },
+                    ebsBlockDevices: [],
+                };
+            }
+
+            // Bottlerocket has two block devices, the root device stores the OS itself and the other is for data like images, logs, persistent storage
```

Comment:
I'm assuming that the root device is not something that can be configured by the user?

Reply:
No it's not. That's configured in the AMI